text/0000-9t_2b_encoding.md

```diff
@@ -0,0 +1,90 @@
+- **(almost) no conversions**: since this encoding plays nicely with byte-wise data processing, bytes as received over the wire can be stored as-is in memory without any prior conversion involving memory allocations and copy operations to receive the converted representation 
+- **partial deserialization**: enables fast deserialization of single transaction fields by simply pointing into a single array with integer offsets and field lengths, thereby preventing costly memory allocations for each transaction field.
+
+# Detailed design
+
+TODO: FINISH THIS SECTION
+
+**Example transaction implementation based on 9/2 encoding**: This example assumes that a trivial trimming algorithm is used that only trims the 0 bytes from the signature message fragment.
+
+```Rust
+const FULL_TRANSACTON_BYTE_SIZE: usize = 1782;
+const ADDRESS_BYTE_OFFSET: usize = 1512;
+const ADDRESS_BYTE_LENGTH: usize = 54;
+
+struct Transaction {
+    bytes: Trits_9enc2, // a type that represents 9trits/2bytes encoded trits
+    offset: usize,
+}
+
+impl Transaction {
+    fn new(bytes: Trits_9enc2) -> Self {
+        Self {
+            bytes,
+            offset: FULL_TRANSACTION_BYTE_SIZE - bytes.len(),
+        }
+    }
+    fn address(&self) -> [Tryte; 81] {
+        //now access and deserialize the address from self.bytes
+        let offset = ADDRESS_BYTE_OFFSET - self.offset;
+        let address = convert::trytes::from_trits_9enc2(&self.bytes[offset..offset+ADDRESS_BYTE_LENGTH]);
+
+        address
+    }
+    fn value(&self) -> u64 {}
+    fn trunk(&self) -> [Tryte; 81] {}
+    fn branch(&self) -> [Tryte; 81] {}
+    fn tag(&self) -> [Tryte; 27] {}
+    ...
+}
+```
+# Drawbacks
+
+Bee needs to be able to integrate with the current IOTA mainnet which employs the `10/2` encoding for the transport, which - as explained earlier - prioritizes memory efficiency over CPU load reduction. If the internal representation of the Bees uses `9/2` then IRI neighbors will become a higher burden for a Bee node, since it has to convert the IRI encoding into `9/2`.
+
+# Rationale and alternatives
+    
+There are basically only 2 options that make sense. Fully use `10/2` and be most memory efficient or fully use `9/2` and be most CPU efficient. During the transition phase  towards the new node software Bee will have to support both encodings at the same time. `10/2` is not optimal in a world of byte-wise data processing. It is more memory efficient, but while bandwidth available to a device can increase externally, the processing capabilities of a single device is predefined and unchangeable. That's why the encoding which optimizes for CPU load reduction should be preferred.
+
+
+# Unresolved questions
+
+TODO
```

Comment:
Could you provide how to interact with hashing? Like could it do operations directly or need some conversions.